weave/integrations/openai/openai_sdk.py

```diff
@@ -0,0 +1,275 @@
+import importlib
+import typing
+from functools import wraps
+
+import weave
+from weave.trace.op_extensions.accumulator import add_accumulator
+from weave.trace.patcher import MultiPatcher, SymbolPatcher
+
+if typing.TYPE_CHECKING:
+    from openai.types.chat import ChatCompletion, ChatCompletionChunk
+
+
+def openai_on_finish_post_processor(value: "ChatCompletionChunk") -> typing.Dict:
+    from openai.types.chat import ChatCompletion, ChatCompletionChunk
+    from openai.types.chat.chat_completion_chunk import (
+        ChoiceDeltaFunctionCall,
+        ChoiceDeltaToolCall,
+    )
+    from openai.types.chat.chat_completion_message import FunctionCall
+    from openai.types.chat.chat_completion_message_tool_call import (
+        ChatCompletionMessageToolCall,
+        Function,
+    )
+
+    def _get_function_call(
+        function_call: typing.Optional[ChoiceDeltaFunctionCall],
+    ) -> typing.Optional[FunctionCall]:
+        if function_call is None:
+            return function_call
+        if isinstance(function_call, ChoiceDeltaFunctionCall):
+            return FunctionCall(
+                arguments=function_call.arguments,
+                name=function_call.name,
+            )
+        else:
+            return None
+
+    def _get_tool_calls(
+        tool_calls: typing.Optional[typing.List[ChoiceDeltaToolCall]],
+    ) -> typing.Optional[typing.List[ChatCompletionMessageToolCall]]:
+        if tool_calls is None:
+            return tool_calls
+
+        _tool_calls = []
+        if isinstance(tool_calls, list):
+            for tool_call in tool_calls:
+                assert isinstance(tool_call, ChoiceDeltaToolCall)
+                _tool_calls.append(
+                    ChatCompletionMessageToolCall(
+                        id=tool_call.id,
+                        type=tool_call.type,
+                        function=Function(
+                            name=tool_call.function.name,
+                            arguments=tool_call.function.arguments,
+                        ),
+                    )
+                )
+        return _tool_calls
+
+    if isinstance(value, ChatCompletionChunk):
```

Comment:
when is `value` NOT a ChatCompletionChunk? the input type should be a union in such cases

Reply:
thanks for catching this.